## 6. Model Specification

**Exercise 6.1**.  Verify Equation (6.1.3) on page 110 for the white noise process.

**Solution**.  Equation (6.1.3) states that, for the white noise process, we have

$$ \text{Var}[r_k] \approx \frac{1}{n} \quad \text{and} \quad \text{Corr}[r_k, r_j] \approx 0 \quad \text{for } k \neq j $$

Equation (6.1.2) states that the joint distribution of

$$ \sqrt{n}(r_1 - \rho_1), \sqrt{n}(r_2 - \rho_2), \cdots, \sqrt{n}(r_m - \rho_m) $$

approaches, as $n \rightarrow \infty$, a joint normal distribution with variances $c_{ii}$ and covariances $c_{ij}$, where

$$ c_{ij} = \sum_{k=-\infty}^\infty \rho_{k+i} \rho_{k+j} + \rho_{k-i} \rho_{k+j} - 2 \rho_{i} \rho_{k} \rho_{k+j} - 2 \rho_{j} \rho_{k} \rho_{k+i} + 2 \rho_{i} \rho_{j} \rho_{k}^2 $$

For a white noise process, $\rho_0 = 1$ and $\rho_a = 0$ for $a \neq 0$, so the equation above reduces to only having a non-zero term when $k = -i = -j$, so $c_{ii} = \rho_0^2 = 1$ and $c_{ij} = 0$ for $i \neq j$.  So the above reduces to  $\sqrt{n}r_k$ having variance 1 and covariance 0, from which Equation (6.1.3) follows.

**Exercise 6.2**.  Verify Equation (6.1.4) on page 110 for the AR(1) process.

**Solution**.  The AR(1) process satisfies $\rho_k = \phi^{|k|}$.  Therefore, for $j \geq 1$, we have from Equation (6.1.2)

$$
\begin{align}
c_{jj} &= \sum_{k=-\infty}^\infty \rho_{k+j}^2 + \rho_{k-j} \rho_{k+j} - 4 \rho_{j} \rho_{k} \rho_{k+j} + 2 \rho_j^2 \rho_{k}^2 \\
&= (1 + 2 \rho_j^2) \sum_{k=-\infty}^\infty \rho_k^2 + \sum_{k=-\infty}^\infty \rho_{k-j} \rho_{k+j} -4 \rho_j \sum_{k=-\infty}^\infty \rho_{k} \rho_{k+j}
\end{align}
$$

Performing each of the geometric sums alluded to in the text, the first sum is

$$
\sum_{k=-\infty}^\infty \rho_k^2 = 1 + 2 \sum_{k=1}^\infty \phi^{2k} = 1 + 2 \frac{\phi^2}{1 - \phi^2} = \frac{1 + \phi^2}{1 - \phi^2}
$$

The second sum is

$$
\begin{align}
\sum_{k=-\infty}^\infty \rho_{k-j} \rho_{k+j} 
&= \sum_{k=-\infty}^{-j-1} \rho_{k-j} \rho_{k+j} + \sum_{k=-j}^{j-1} \rho_{k-j} \rho_{k+j} + \sum_{k=j}^\infty \rho_{k-j} \rho_{k+j} \\
&= \sum_{k=-\infty}^{-j-1} \phi^{-(k-j)} \phi^{-(k+j)} + \sum_{k=-j}^{j-1} \phi^{k-j} \phi^{-(k+j)} + \sum_{k=j}^\infty \phi^{k-j} \phi^{k+j} \\
&= \sum_{k=-\infty}^{-j-1} \phi^{-2k} + \sum_{k=-j}^{j-1} \phi^{-2j} + \sum_{k=j}^\infty \phi^{2k} \\
&= \frac{\phi^{2j+2}}{1 - \phi^2} + 2j \phi^{2j} + \frac{\phi^{2j}}{1 - \phi^2} \\
&= \phi^{2j} \left( 2j + \frac{1 + \phi^2}{1 - \phi^2} \right)
\end{align}
$$

The third sum is

$$
\begin{align}
\sum_{k=-\infty}^\infty \rho_{k} \rho_{k+j} 
&= \sum_{k=-\infty}^{-j} \rho_{k} \rho_{k+j} + \sum_{k=-j+1}^{0} \rho_{k} \rho_{k+j} + \sum_{k=1}^\infty \rho_{k} \rho_{k+j} \\
&= \sum_{k=-\infty}^{-j} \phi^{-k} \phi^{-(k+j)} + \sum_{k=-j+1}^{0} \phi^{-k} \phi^{k+j} + \sum_{k=1}^\infty \phi^{k} \phi^{k+j} \\
&= \phi^{-j} \sum_{k=-\infty}^{-j-1} \phi^{-2k} + \sum_{k=-j+1}^{0} \phi^{j} + \phi^j \sum_{k=1}^\infty \phi^{2k} \\
&= \phi^{-j} \frac{\phi^{2j}}{1 - \phi^2} + j \phi^j + \phi^j \frac{\phi^2}{1 - \phi^2} \\
&= \phi^j \left( j + \frac{1 + \phi^2}{1 - \phi^2} \right)
\end{align}
$$

Replacing on the original expression,

$$\begin{align} 
c_{jj} &= (1 + 2 \phi^{2j}) \frac{1 + \phi^2}{1 - \phi^2} + \phi^{2j} \left( 2j + \frac{1 + \phi^2}{1 - \phi^2} \right) -4 \phi^{2j} \left( j + \frac{1 + \phi^2}{1 - \phi^2} \right) \\
&= (1 - \phi^{2j})\frac{1 + \phi^2}{1 - \phi^2} - 2j\phi^{2j}
\end{align}
$$

Since this refers to the limit of the variance of $\sqrt{n} r_j$, we get the result

$$ \text{Var}[r_k] \approx \frac{1}{n}\left[ (1 - \phi^{2k})\frac{1 + \phi^2}{1 - \phi^2} - 2k \phi^{2k} \right] $$

**Exercise 6.3**.  Verify the line in Exhibit 6.1 on page 111, for values $\phi = \pm 0.9$.

**Solution**.  We will use Equation (6.1.2) for $k = 1, 2, 10$:

$$ \text{Var}[r_k] \approx \frac{1}{n}\left[ (1 - \phi^{2k})\frac{1 + \phi^2}{1 - \phi^2} - 2k \phi^{2k} \right] $$

and Equation (6.1.8):

$$ \text{Corr}[r_1, r_2] \approx 2 \phi \sqrt{\frac{1 - \phi^2}{1 + 2\phi^2 - 3\phi^4}} $$

In [1]:
import numpy as np

def se_rk(phi, k):
    return np.sqrt((1 - phi**(2*k)) * (1 + phi**2) / (1 - phi**2) - 2 * k * phi**(2*k))

def corr(phi):
    return 2 * phi * np.sqrt((1 - phi**2)/(1 + 2 * phi**2 - 3 * phi**4))

In [2]:
phi = 0.9

for value in [se_rk(phi, 1), se_rk(phi, 2), corr(phi), se_rk(phi, 10)]:
    print(round(value, 2))

0.44
0.81
0.97
2.44


**Exercise 6.4**.  Add new entries to Exhibit 6.1 on page 111, for the following values:

**(a)** $\phi = \pm 0.99$

**(b)** $\phi = \pm 0.5$

**(c)** $\phi = \pm 0.1$

**Solution**.

In [3]:
phi = 0.99

for value in [se_rk(phi, 1), se_rk(phi, 2), corr(phi), se_rk(phi, 10)]:
    print(round(value, 2))

0.14
0.28
1.0
1.33


In [4]:
phi = 0.5

for value in [se_rk(phi, 1), se_rk(phi, 2), corr(phi), se_rk(phi, 10)]:
    print(round(value, 2))

0.87
1.15
0.76
1.29


In [5]:
phi = 0.1

for value in [se_rk(phi, 1), se_rk(phi, 2), corr(phi), se_rk(phi, 10)]:
    print(round(value, 2))

0.99
1.01
0.2
1.01


$$
\begin{array}{ccccc}
\phi & \sqrt{\text{Var}[r_1]} & \sqrt{\text{Var}[r_2]} & \text{Corr}[r_1, r_2] & \sqrt{\text{Var}[r_{10}]} \\
\pm 0.99 & 0.14 / \sqrt{n} & 0.28 / \sqrt{n} & \pm 1.00 & 1.33 / \sqrt{n} \\
\pm 0.9  & 0.44 / \sqrt{n} & 0.81 / \sqrt{n} & \pm 0.97 & 2.44 / \sqrt{n} \\
\pm 0.7  & 0.71 / \sqrt{n} & 1.12 / \sqrt{n} & \pm 0.89 & 1.70 / \sqrt{n} \\
\pm 0.5  & 0.87 / \sqrt{n} & 1.15 / \sqrt{n} & \pm 0.76 & 1.29 / \sqrt{n} \\
\pm 0.4  & 0.92 / \sqrt{n} & 1.11 / \sqrt{n} & \pm 0.66 & 1.18 / \sqrt{n} \\
\pm 0.2  & 0.98 / \sqrt{n} & 1.04 / \sqrt{n} & \pm 0.38 & 1.04 / \sqrt{n} \\
\pm 0.1  & 0.99 / \sqrt{n} & 1.01 / \sqrt{n} & \pm 0.20 & 1.01 / \sqrt{n}
\end{array}
$$

**Exercise 6.5**.  Verify Equation (6.1.9) on page 111 and Equation (6.1.10) for the MA(1) process.

**Solution**.  Equation (6.1.2) is that

$$ c_{ij} = \sum_{k=-\infty}^\infty \rho_{k+i} \rho_{k+j} + \rho_{k-i} \rho_{k+j} - 2 \rho_{i} \rho_{k} \rho_{k+j} - 2 \rho_{j} \rho_{k} \rho_{k+i} + 2 \rho_{i} \rho_{j} \rho_{k}^2 $$

and, in particular for $i = j$, we get

$$ c_{jj} = \sum_{k=-\infty}^\infty \rho_{k+j}^2 + \rho_{k-j} \rho_{k+j} - 4 \rho_{j} \rho_{k} \rho_{k+j} + 2 \rho_j^2 \rho_{k}^2 $$

Equation (6.1.9) states that, for a MA(1) process,

$$ c_{11} = 1 - 3 \rho_1^2 + 4 \rho_1^4 \quad \text{and} \quad c_{kk} = 1 + 2\rho_1^2 \quad \text{for } k > 1 $$

But for the MA(1) process, the autocorrelation function is

$$ \rho_k = \begin{cases}
1 &\text{for } k = 0 \\
\theta &\text{for } |k| = 1 \\
0 &\text{otherwise}
\end{cases} $$

For $c_{11}$, the sums decompose as

- $ \sum_{k=-\infty}^\infty \rho_{k+j}^2 = \rho_{-1}^2 + \rho_0^2 + \rho_1^2 = 1 + 2\theta^2 $
- $ \sum_{k=-\infty}^\infty \rho_{k-j} \rho_{k+j} = \rho_{-1} \rho_1 = \theta^2 $
- $ \sum_{k=-\infty}^\infty \rho_{j} \rho_{k} \rho_{k+j} = \rho_{-1} \rho_0 \rho_1 = \theta^2 $
- $ \sum_{k=-\infty}^\infty \rho_j^2 \rho_{k}^2 = \theta^4 $

and so we get the first half of Equation (6.1.9),

$$ c_{11} = 1 + 2 \theta^2 + \theta^2 - 4 \theta^2 + 4 \theta^4 = 1 - 3 \theta^2 + 4 \theta^4 $$

For $c_{jj}$, with $j > 1$, the sums decompose as

- $ \sum_{k=-\infty}^\infty \rho_{k+j}^2 = \rho_{-1}^2 + \rho_0^2 + \rho_1^2 = 1 + 2\theta^2 $
- $ \sum_{k=-\infty}^\infty \rho_{k-j} \rho_{k+j} = 0 $
- $ \sum_{k=-\infty}^\infty \rho_{j} \rho_{k} \rho_{k+j} = 0 $
- $ \sum_{k=-\infty}^\infty \rho_j^2 \rho_{k}^2 = 0$

and so we get the second half of Equation (6.1.9),

$$ c_{jj} = 1 + 2 \theta^2 $$

Equation (6.1.10) states that 

$$ c_{12} = 2 \rho_1 (1 - \rho_1^2) $$

But we have

$$ 
\begin{align}
c_{12} &= \sum_{k=-\infty}^\infty \rho_{k+1} \rho_{k+2} + \rho_{k-1} \rho_{k+2} - 2 \rho_{1} \rho_{k} \rho_{k+2} - 2 \rho_{2} \rho_{k} \rho_{k+1} + 2 \rho_{1} \rho_{2} \rho_{k}^2 \\
&= \rho_{-1}\rho_0 + \rho_0 \rho_1 - 2\rho_1 \rho_{-1} \rho_{1} \\
&= \theta + \theta - 2 \theta^3 \\
&= 2 \theta (1 - \theta^3)
\end{align}
$$